src/cogs/snipe.py

```diff
@@ -25,11 +25,18 @@ def __init__(self, bot):
         self._message_history: dict[int, list[discord.Message]] = defaultdict(lambda: deque(maxlen=conf.max_edit_msg))
 
     @commands.hybrid_command(with_app_command=True)
-    async def snipe(self, ctx: Context, number: int = 0):
-        if ctx.channel.id not in self._deleted:
-            return await ctx.send("No message to snipe.")
+    async def snipe(self, ctx: Context, number: int = 0, channel: int = 0):
+        if channel == 0:
```

Comment:
Please use an Optional, default it to none and then check for that. This also ends up working better when using slash commands. Furthermore, if you use a `Channel` object instead of an `int` the slash command will let you pick the channel directly, which would be good seeing how large of a hassle getting the id of a channel is in comparison. But I'm not aware of how that ends up interacting with text commands, although I believe worst case you'd just have an optional parameter you couldn't use.

Reply:
@krisvers he's right